### PR TITLE
Docker troubleshooting windows

### PR DESCRIPTION
**Pull request by KhalidUmuzi**
Related issues: [please specify]

## Description:

New Docker Troubleshooting content.

## I solemnly swear that:

- [X] I ran the hugo server and looked at my changed in the browser with my own eyes
- [x] I ran the linter and there were no errors
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code

### CODE REVIEW

File: content/docker/docker-desktop-troubleshooting/_index.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## Virtualization
+If you are still getting some errors, your PC might have virtualization disabled. In order to run docker, or rather, any Virtual Machine on your Local Machine, you need to enable virtualization. Unfortunately this involves you going into the BIOS and enabling it there. The method can differ depending on what hardware you are using and you may need to delve a little deeper.
+
+Here is a basic [link](https://www.minitool.com/news/enable-virtualization-windows-10.html) to get you started. If this method does not work for you, you will need to google your own hardware in order to enable virtualization on your machine.
+
+## None of these helped me. I'm still stuck!
+At this point we would strongly advise against Docker Desktop on Windows since it may not be compatible with your current setup. Your alternatives are
+
+1.) Use WSL as mentioned in the intro to docker Topic. Follow the instructions to install WSL and docker very carefully as missing one step may lead to Docker not working as expected.
```
Comment (Andy-Nkumane) [inline content/docker/docker-desktop-troubleshooting/_index.md]:
```suggestion
1.) Use WSL 2 as mentioned in the intro to docker Topic. Follow the instructions to install WSL 2 and docker very carefully as missing one step may lead to Docker not working as expected.
```

Comment (Andy-Nkumane) [reply content/docker/docker-desktop-troubleshooting/_index.md]: having accurate details removes any confusion that may arise

File: content/docker/docker-desktop-troubleshooting/_index.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+2.) Installing a Linux Virtual Machine is an alternative as well. There are many free tools that allow for this such as [VMware](https://www.vmware.com/products/workstation-player/workstation-player-evaluation.html). You will need to also download a Linux Operating System for it to use as well. Linux is open source, so this shouldnt be hard to find with some searching online. An easy search term for example would be "Ubuntu vmware image download".
```
Comment (Andy-Nkumane) [inline content/docker/docker-desktop-troubleshooting/_index.md]: is a virtual machine a viable option for everyone? i know that some if not all virtual machine software requires a lot of computational power and not everyone may have enough to run it smoothly which will result in slow-performing devices and that is something we don't want. maybe adding the option of having a linux subsystem may be viable, what do you think?

**Review by Andy-Nkumane — CHANGES_REQUESTED** [review]
here are my thoughts on this:
* if this document recommends a better way to install docker on windows than the one mentioned in the intro to docker document - wouldn't it make sense to move all the instructions on installing docker for windows to this document and have them in the preferred order

File: content/docker/docker-desktop-troubleshooting/_index.md (hunk 2 above)
Comment (KhalidUmuzi) [reply content/docker/docker-desktop-troubleshooting/_index.md]: So for this one, I think having it as an option would be fine for those who have the means to do so. The Linux Subsystem is mentioned above. the VM is the absolute last resort(I think ill change the wording to reflect this).

**Comment by KhalidUmuzi** [comment]
> 

So this is for those who have followed the instructions in the intro topic correctly and are still struggling. The Intro Topic is the normal way of installing. The new topic isn't better in any way, unless you are having issues with Docker Desktop after following instructions. The steps outlined in the topic can be done after Everything in the intro topic has been done correctly and the student is still having an issue.

What do you think?

**Review by Andy-Nkumane — APPROVED** [review]

File: content/docker/docker-desktop-troubleshooting/_index.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+Once this is done, it should allow you to run Docker Desktop without giving you any errors about Hyper-V
+
+## Virtualization
+If you are still getting some errors, your PC might have virtualization disabled. In order to run docker, or rather, any Virtual Machine on your Local Machine, you need to enable virtualization. Unfortunately this involves you going into the BIOS and enabling it there. The method can differ depending on what hardware you are using and you may need to delve a little deeper.
+
+Here is a basic [link](https://www.minitool.com/news/enable-virtualization-windows-10.html) to get you started. If this method does not work for you, you will need to google your own hardware in order to enable virtualization on your machine.
+
+## None of these helped me. I'm still stuck!
+At this point Docker Desktop on Windows may not be compatible with your current setup. Your alternatives are
```
Comment (ry-oc) [inline content/docker/docker-desktop-troubleshooting/_index.md]: `Your alternatives are:` should be on a new line, it does not look right where it is

**Review by ry-oc — CHANGES_REQUESTED** [review]
Minor formatting change, also this should be added to peoples syllabuses in the appropriate place to ensure they get it at the correct time instead of floundering.

**Review by Vuyisanani — APPROVED** [review]

**Review by Vuyisanani — CHANGES_REQUESTED** [review]
Please resolve your merge conflicts

**Review by Andy-Nkumane — APPROVED** [review]

**Review by ry-oc — APPROVED** [review]

File: content/sql/shop-project/_index.md [hunk 4]
```diff
@@ -14,6 +14,7 @@ prerequisites:
   hard:
   - topics/solo-learn/sql/4-challenges
   - docker/intro-to-docker/
+  - docker/docker-desktop-troubleshooting
```
Comment (sheenarbw) [inline content/sql/shop-project/_index.md]: Why is this one a hard-prereq but everywhere else it's soft?

Would it be better to just link to the troubleshooting from the "intro-to-docker" content? 

It doesn't necessarily need to show up as a card on people's boards even, you can just link to it from inside the intro content.

Something like:

```
## Running docker on windows

If you are running Windows then getting docker to run can be a bit tricky.  The following link will help you get set up: 

...
```

**Review by sheenarbw — CHANGES_REQUESTED** [review]
This is good but there are some inconsistencies in hard/soft prerequisites. I suggested a simplification in the comments.

File: content/sql/shop-project/_index.md (hunk 4 above)
Comment (KhalidUmuzi) [reply content/sql/shop-project/_index.md]: Addressed. link to topic already in the intro topic on line 61

**Review by ry-oc — APPROVED** [review]

File: content/docker/docker-desktop-troubleshooting/_index.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ready: false
+title: Troubleshooting Docker Desktop for Windows
+---
+
+Docker on Windows can be quite tricky to setup and get working, and a number of errors can occur when trying to get it up and running. In this topic, we will try and provide solutions to some of the problems you may encounter when dealing with Docker Desktop.
+
+## Hyper-V
+Docker Desktop requires Hyper-V to be installed as well as enabled on your machine. This installation is in the form of a feature. Click [here](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v) and follow the instructions to ensure Hyper-V is enabled on your machine.
+
+### Help! I dont see Hyper-V on my list of Programs and features!
```
Comment (sheenarbw) [inline content/docker/docker-desktop-troubleshooting/_index.md]: inconsistent capitalisation. Thy does Programs get a capital and features doesn't?

Just use regular sentence-case in the headings.

File: content/docker/docker-desktop-troubleshooting/_index.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+Now save that as hyperv.bat from your text editor. The desktop or download folder are good destinations for which to save and easily located for the next step.
+
+After exiting the editor, open file manager and right mouse click the file and choose run as administrator.
+
+Once complete it will reboot your machine and you can turn on Hyper-v from programs and features.
+
+Once this is done, it should allow you to run Docker Desktop without giving you any errors about Hyper-V
+
+## Virtualization
```
Comment (sheenarbw) [inline content/docker/docker-desktop-troubleshooting/_index.md:32]: please leave newline characters under headings

File: content/docker/flask-and-docker/_index.md [hunk 7]
```diff
@@ -9,12 +9,14 @@ prerequisites:
   - docker/intro-to-docker
   - projects/python-specific/flask-rest-api/part-1
   - projects/python-specific/flask-rest-api/part-2
+  soft:
```
Comment (sheenarbw) [inline content/docker/flask-and-docker/_index.md]: please remove the docker/docker-desktop-troubleshooting prerequisite. It should only be in the intro to docker content

File: content/docker/docker-desktop-troubleshooting/_index.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+---
+content_type: topic
+prerequisites:
+  hard:
+  - docker/intro-to-docker
+ready: false
```
Comment (sheenarbw) [inline content/docker/docker-desktop-troubleshooting/_index.md]: ready: true.

Unless this actually is incomplete?

File: content/docker/intro-to-docker/_index.md [hunk 9]
```diff
@@ -25,38 +25,40 @@ In this section, we'll get docker set up on your computer. Then we'll use it to
...
-https://docs.docker.com/engine/install/linux-postinstall/
+https://docs.docker.com/desktop/windows/install/
 
-### Now install docker-compose
+An extension for VSCode can be installed as well https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker
 
-https://docs.docker.com/compose/install/
+You wont need to install docker-compose as this is already included.
+
+If you are having issues with Docker Desktop, please see this topic for troubleshooting {{% contentlink path="docker/docker-desktop-troubleshooting" %}}
```
Comment (sheenarbw) [inline content/docker/intro-to-docker/_index.md:61]: This is the best place to put this.

File: content/projects/java-specific/collections/_index.md [hunk 10]
```diff
@@ -9,6 +9,7 @@ prerequisites:
   - topics/java-specific/collections-and-datastructures
   soft:
   - docker/intro-to-docker
+  - docker/docker-desktop-troubleshooting
```
Comment (sheenarbw) [inline content/projects/java-specific/collections/_index.md]: This is already mentioned in intro-to-docker. Please remove the extra prerequisite

**Review by sheenarbw — CHANGES_REQUESTED** [review]
There's some tidying to be done and extra prerequisites can be removed

**Review by ry-oc — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**